src/gateway/routers/Aggregator.sol

```diff
@@ -189,15 +204,57 @@ contract Aggregator is Auth, IAggregator {
 
     // --- Outgoing ---
     /// @inheritdoc IAggregator
-    function send(bytes calldata message) external auth {
+    function send(bytes calldata message) external payable auth {
         uint256 numRouters = routers.length;
         require(numRouters > 0, "Aggregator/not-initialized");
 
+        uint256 fuel = msg.value;
+
+        uint256 destChainMsgCost;
+        uint256 destChainProofCost;
+
         bytes memory proof = abi.encodePacked(uint8(MessagesLib.Call.MessageProof), keccak256(message));
+
+        if (fuel > 0) {
+            destChainMsgCost = gasService.estimate(message);
+            destChainProofCost = gasService.estimate(proof);
+        }
+
         for (uint256 i; i < numRouters; i++) {
-            RouterLike(routers[i]).send(i == PRIMARY_ROUTER_ID - 1 ? message : proof);
+            RouterLike currentRouter = RouterLike(routers[i]);
+            bytes memory payload = i == PRIMARY_ROUTER_ID - 1 ? message : proof;
+            // TODO a discussion:
```

Comment:
@hieronx  have you thought about this one?
If you agree we it , let me know so that I can remove this comment

Reply:
Yeah I was still thinking about this a bit more, but for now I think the solution you have gone for makes sense.